### DISCUSSION TITLE
Jax is definitely installed, yet JaxSolver does not run

### DISCUSSION THREAD

**Discussion by cyasing** [pybamm-team/PyBaMM#2722] [Q&A] [answered]
Just as the title says, I can't seem to run JaxSolver() even with Jax installed.

![image](https://user-images.githubusercontent.com/59028917/220842849-9349f833-f99c-4472-915e-4eda3a729752.png)

![image](https://user-images.githubusercontent.com/59028917/220843763-07b23a4a-6b45-4a33-aa08-437ab4e6227c.png)

I tried the pybamm_install_jax, but that gives this: `NameError: name 'pybamm_install_jax' is not defined`. 
What version of Jax is compatible with PyBaMM?

**Comment by Saransh-cpp** [comment, marked as answer]
Supported `jax` and `jaxlib` versions:

https://github.com/pybamm-team/PyBaMM/blob/4f03df423bc63f9b4bd5f7e926c78a2ad1af1f87/pybamm/util.py#L24-L26

I think you are executing `pybamm_install_jax` in a `python` file? `pybamm_install_jax` should be executed in your terminal.

Also, I see that the docstring is a bit misleading. The docstring uses `>>>`, misleading users into executing it in the `python` interpreter (a PR fixing it would be very welcome!) -

https://github.com/pybamm-team/PyBaMM/blob/4f03df423bc63f9b4bd5f7e926c78a2ad1af1f87/pybamm/util.py#L330-L341

**Reply by Saransh-cpp** [reply]
Also, https://pybamm.readthedocs.io/en/latest/source/api/util.html misses the API documentation for `install_jax`. The function can be added here: https://github.com/pybamm-team/PyBaMM/blob/develop/docs/source/api/util.rst.

**Reply by cyasing** [reply]
Thanks! That worked. However, I am encountering this error now: 
` raise TracerArrayConversionError(self)
jax._src.errors.TracerArrayConversionError: The numpy.ndarray conversion method __array__() was called on the JAX Tracer object Traced<ShapedArray(float64[1,1])>with<DynamicJaxprTrace(level=1/2)> (https://jax.readthedocs.io/en/latest/errors.html#jax.errors.TracerArrayConversionError)`

![image](https://user-images.githubusercontent.com/59028917/220858279-658faccc-7abb-4af7-97c8-87373e4d9663.png)

I used `method=BDF`, because for some reason the default one does not work. I had to use 'model.events=[]' to remove events before attempting to solve using BDF.
